scripts/test.sh

```diff
@@ -559,8 +559,15 @@ function dep_pass {
 
 function release_pass {
   rm -f ./bin/etcd-last-release
-  # to grab latest patch release; bump this up for every minor release
-  UPGRADE_VER=$(git tag -l --sort=-version:refname "v3.5.*" | head -1 | cut -d- -f1)
+
+  # This gets a list of all remote tags for the release branch in regex
+  # Sort key is used to sort numerically by patch version
+  # Latest version is then stored for use below
+  UPGRADE_VER=$(git ls-remote --tags https://github.com/etcd-io/etcd.git \
+    | grep --only-matching --perl-regexp '(?<=v)3.5.[\d]+?(?=[\^])' \
```

Comment:
For `main` branch we want to run test against v3.5 minor, but for `release-3.5` we should be picking from v3.4.

Reply:
Yeah I believe we just need to backport this change to `release-3.5` and change the regex to 3.4 once this merges.

Reply:
`make test-e2e-release` always runs `make build` first so how about something like this one liner, no params required, will always stay up to date based on minor version reported by the etcd binary:

```bash
echo $(( $(./bin/etcd --version | grep --only-matching --perl-regexp '(?<=etcd Version: 3.)\d') - 1 ))
```

Reply:
That's a nice solution.

Reply:
Cool ok - added a handler for the edge case of new major version, think it's all sorted now pending tests green ✔️